test-utils/src/main/java/com/revenuecat/purchases/utils/billingClientStubs.kt

```diff
@@ -19,34 +23,85 @@ import org.assertj.core.api.AssertionsForClassTypes.fail
 import org.json.JSONArray
 
 @SuppressWarnings("LongParameterList", "MagicNumber")
-fun stubSkuDetails(
-    productId: String = "monthly_intro_pricing_one_week",
+fun mockProductDetails(
+    productId: String = "sample_product_id",
     @BillingClient.ProductType type: String = BillingClient.ProductType.SUBS,
+    oneTimePurchaseOfferDetails: OneTimePurchaseOfferDetails? = null,
+    subscriptionOfferDetails: List<SubscriptionOfferDetails> = listOf(mockSubscriptionOfferDetails()),
+    name: String = "subscription_mock_name",
+    description: String = "subscription_mock_description",
+    title: String = "subscription_mock_title"
+): ProductDetails = mockk<ProductDetails>().apply {
+    every { getProductId() } returns productId
+    every { productType } returns type
+    every { getName() } returns name
+    every { getDescription() } returns description
+    every { getTitle() } returns title
+    every { getOneTimePurchaseOfferDetails() } returns oneTimePurchaseOfferDetails
+    every { getSubscriptionOfferDetails() } returns subscriptionOfferDetails
+    every { zza() } returns "mock-package-name" // This seems to return the packageName property from the response json
```

Comment:
This is an annoying case... This is part of the public API of `ProductDetails` even though it's obfuscated. Additionally, this seems to be used when creating the `BillingFlowParams` so we need to mock it.

Reply:
hmm i didn't know we used this for `BillingFlowParams`, can you share where you saw that?

Reply:
We don't. That's used internally in Google's implementation of `BillingFlowParams` when we build it. As for what they use it for, no idea since it's obfuscated :(

Reply:
ohhh got it. thanks!